.github/workflows/csharp-test.yml

```diff
@@ -65,26 +67,18 @@ jobs:
 
       - name: Cache Rust dependencies
         uses: Swatinem/rust-cache@v2
-        id: cache-rust-deps
         with:
-          workspaces: demo/Blackholio/server-rust
-          key: ${{ steps.checkout-stdb.outputs.commit }}
-          # Cache Rust deps even if unit tests have failed.
-          cache-on-failure: true
-          # Cache the CLI as well.
-          cache-all-crates: true
+          workspaces: ${{ github.workspace }}
+          shared-key: spacetimedb
+          # Let the main CI job save the cache since it builds the most things
+          save-if: false
 
       - name: Install SpacetimeDB CLI from the local checkout
-        # Rebuild only if we didn't get a precise cache hit.
-        if: steps.cache-rust-deps.outputs.cache-hit == 'false'
```

Comment:
I don't feel 100% confident in this cache-skipping logic, so I removed it. If we hit the cache, this step will be fast anyway.